### DISCUSSION TITLE
Layout bug in GWT client?

### DISCUSSION THREAD

**Discussion by armin-reichert** [webfx-project/webfx#22] [General]
In the WebFX Pac-Man app (https://github.com/webfx-demos/webfx-pacman) you can press Alt+D to show some debug drawing (grid lines, additional borders). In the desktop client this looks fine but in the browser it looks different and wrong, see attached pictures.

![desktop](https://github.com/webfx-project/webfx/assets/7394275/f3941486-45ee-4c06-87ab-d00181071acf)

![browser](https://github.com/webfx-project/webfx/assets/7394275/bdac7425-0a84-4f81-99bc-fd091c881279)

Update: Using the browser dev tools I found out that this is not a layout issue but that the wallpaper image seems to overpaint the borders.

**Comment by salmonb** [comment]
hi Armin, there is a little issue with borders indeed in WebFX. This is due to the difference in the way JavaFX and HTML manage borders: when you set a border in HTML, let's say of 5px width, this shifts the background which will then start at (5, 5), while JavaFX doesn't shift the background but keep it starting at (0, 0). This difference causes some side-effects like those you reported.

I will try to fix that at some point, but it's not in my priorities at the moment. Hope it's ok.

**Comment by salmonb** [comment]
hi Armin,

I finally found a bit of time to work on this...

I think the latest WebFX version fixes this issue.

Can you please confirm?

**Comment by armin-reichert** [comment]
Hi Bruno,

I tried to trigger a rebuild of https://github.com/webfx-demos/webfx-pacman (by removing some garbage) but this rebuild failed. I have no idea what the problem is.

**Reply by salmonb** [reply]
Maybe a 'webfx update' is required, can you please check?

**Reply by salmonb** [reply]
Also make sure to have the latest version of the WebFX CLI. So can you run these commands along your local repo:

- `webfx bump cli`
- `webfx update`
- `webfx build -g`

If your local build is successful, can you please push the changes made by the CLI to the GitHub repo?

**Comment by armin-reichert** [comment]
BTW: I think we should delete the old projects `webfx-demo-pacman` and `webfx-demo-pacman-core`.

**Comment by armin-reichert** [comment]
I did what you told. Local build works fine. But there are no changes in the
webfx.xml files (as far as IntelliJ tells me). 
 
Yesterday, I just removed some commented code on GitHub to trigger a central
build of project webfx-pacman and this build failed. Any idea why?
 
Regards
Armin

  _____  

Von: Bruno Salmon ***@***.*** 
Gesendet: Dienstag, 14. November 2023 10:23
An: webfx-project/webfx
Cc: Armin Reichert; Author
Betreff: Re: [webfx-project/webfx] Layout bug in GWT client? (Discussion
#22)



Also make sure to have the latest version of the WebFX CLI. So can you run
these commands along your local repo:

*	webfx bump cli 

*	webfx update 

*	webfx build -g 

If your local build is successful, can you please push the changes made by
the CLI to the GitHub repo?

-
Reply to this email directly, view
<https://github.com/webfx-project/webfx/discussions/22#discussioncomment-756
3252> it on GitHub, or unsubscribe
<https://github.com/notifications/unsubscribe-auth/ABYNHY46FEYA6GKJLWHUMI3YE
MZ55AVCNFSM6AAAAAA2ELLJG6VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TK
NRTGI2TE> .
You are receiving this because you authored the thread.
<https://github.com/notifications/beacon/ABYNHY4VVE66ZHD3TRDPGIDYEMZ55A5CNFS
M6AAAAAA2ELLJG6WGG33NNVSW45C7OR4XAZNRIRUXGY3VONZWS33OINXW23LFNZ2KUY3PNVWWK3T
UL5UWJTQAONT7I.gif> Message ID:
***@***.***>

**Reply by salmonb** [reply]
The changes are probably not in the webfx.xml files (which are the input of the CLI together with the source code), but rather in the build chain (the output), such as pom.xml or module.gwt.xml, module-info.java, etc...

Are you sure none of these files have been updated?
IntelliJ should display these changes in its "Local changes" view.

**Comment by armin-reichert** [comment]
Yes, no changes at all.
 
I would expect to see any changed files (which are not git-ignored) in the
"Commit" view of IntelliJ, or not? A "Local changes" view seems not to exist
in my IntelliJ version (which is the latest)

  _____  

Von: Bruno Salmon ***@***.*** 
Gesendet: Dienstag, 14. November 2023 10:38
An: webfx-project/webfx
Cc: Armin Reichert; Author
Betreff: Re: [webfx-project/webfx] Layout bug in GWT client? (Discussion
#22)



The changes are probably not in the webfx.xml files (which are the input of
the CLI together with the source code), but rather in the build chain (the
output), such as pom.xml or module.gwt.xml, module-info.java, etc...

Are you sure none of these files have been updated?
IntelliJ should display these changes in its "Local changes" view.

-
Reply to this email directly, view
<https://github.com/webfx-project/webfx/discussions/22#discussioncomment-756
3421> it on GitHub, or unsubscribe
<https://github.com/notifications/unsubscribe-auth/ABYNHY6QREBJNL32LKZXIGDYE
M3W3AVCNFSM6AAAAAA2ELLJG6VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TK
NRTGQZDC> .
You are receiving this because you authored the thread.
<https://github.com/notifications/beacon/ABYNHY3ZTUMKTTKKA3LRN6TYEM3W3A5CNFS
M6AAAAAA2ELLJG6WGG33NNVSW45C7OR4XAZNRIRUXGY3VONZWS33OINXW23LFNZ2KUY3PNVWWK3T
UL5UWJTQAONUJ2.gif> Message ID:
***@***.***>

**Reply by salmonb** [reply]
I have the latest IntelliJ too, but kept the old VCS layout... In the new layout, it should look like this:

![image](https://resources.jetbrains.com/help/img/idea/2023.2/ij_diff_in_editor.jpg)

So nothing to commit in your "Default Changelist"?

**Comment by armin-reichert** [comment]
That's how it looks for me:
 


  _____  

Von: Bruno Salmon ***@***.*** 
Gesendet: Dienstag, 14. November 2023 10:51
An: webfx-project/webfx
Cc: Armin Reichert; Author
Betreff: Re: [webfx-project/webfx] Layout bug in GWT client? (Discussion
#22)



I have the latest IntelliJ too, but kept the old CSV layout... In the new
layout, it should look like this:

 
<https://camo.githubusercontent.com/ca7302e6501bec90ab70ec8a24e1430a7ebb1f40
4a2a66da65ff190b216fafdd/68747470733a2f2f7265736f75726365732e6a6574627261696
e732e636f6d2f68656c702f696d672f696465612f323032332e322f696a5f646966665f696e5
f656469746f722e6a7067> image

So nothing to commit in your "Default Changelist"?

-
Reply to this email directly, view
<https://github.com/webfx-project/webfx/discussions/22#discussioncomment-756
3583> it on GitHub, or unsubscribe
<https://github.com/notifications/unsubscribe-auth/ABYNHY2642DMT5RCUZ2LGHTYE
M5JRAVCNFSM6AAAAAA2ELLJG6VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TK
NRTGU4DG> .
You are receiving this because you authored the thread.
<https://github.com/notifications/beacon/ABYNHYYY7SSCC52HAOJR4H3YEM5JRA5CNFS
M6AAAAAA2ELLJG6WGG33NNVSW45C7OR4XAZNRIRUXGY3VONZWS33OINXW23LFNZ2KUY3PNVWWK3T
UL5UWJTQAONUT6.gif> Message ID:
***@***.***>

**Reply by salmonb** [reply]
Did you include an image? can't see it...

I updated my local repo with your latest pacman version and ran `webfx update`.
This updated the build chain (7 files changed).
I pushed that to the GitHub repo, and the GitHub workflow ran successfully.
So [https://pacman.webfx.dev](https://pacman.webfx.dev) and [https://mspacman.webfx.dev](https://mspacman.webfx.dev) have been updated.

I don't know why that didn't work on your machine... Maybe some old snapshots in your local repo that Maven didn't refresh for any reason 🤷

**Comment by armin-reichert** [comment]
Yes I pasted a screenshot which I can see in the sent mail. WTF.
 
So I will clean my local Maven cache and try again.
 
Thanks for your help!

  _____  

Von: Bruno Salmon ***@***.*** 
Gesendet: Dienstag, 14. November 2023 11:38
An: webfx-project/webfx
Cc: Armin Reichert; Author
Betreff: Re: [webfx-project/webfx] Layout bug in GWT client? (Discussion #22)



Did you include an image? can't see it...

I updated my local repo with your latest pacman version and ran webfx update.
This updated the build chain (7 files changed).
I pushed that to the GitHub repo, and the GitHub workflow ran successfully.
So https://pacman.webfx.dev and https://mspacman.webfx.dev have been updated.

I don't know why that didn't work on your machine... Maybe some old snapshots in your local repo that Maven didn't refresh for any reason 🤷

**Comment by armin-reichert** [comment]
As you can see in the picture, the yellow and green borders look fine now in the GWT UI. But the red border is missing. The right window shows the desktop version with all three borders displayed correctly. However, these borders are only used for debugging and have no use for the game, so you may close this issue if you want.

![screenshot](https://github.com/webfx-project/webfx/assets/7394275/76217e4e-7455-4b87-803d-79c6de9ee7af)

**Reply by salmonb** [reply]
Thanks for spotting this. I think I have fixed this issue in WebFX as well now. Can you confirm?

**Comment by armin-reichert** [comment]
Indeed. Now it looks perfect on https://pacman.webfx.dev/ (Press Alt+D to
verify)

  _____  

Von: Bruno Salmon ***@***.*** 
Gesendet: Dienstag, 14. November 2023 15:51
An: webfx-project/webfx
Cc: Armin Reichert; Author
Betreff: Re: [webfx-project/webfx] Layout bug in GWT client? (Discussion
#22)



Thanks for spotting this. I think I have fixed this issue in WebFX as well
now. Can you confirm?

-
Reply to this email directly, view
<https://github.com/webfx-project/webfx/discussions/22#discussioncomment-756
6457> it on GitHub, or unsubscribe
<https://github.com/notifications/unsubscribe-auth/ABYNHY25WA6WNZAYONNN5IDYE
OANXAVCNFSM6AAAAAA2ELLJG6VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TK
NRWGQ2TO> .
You are receiving this because you authored the thread.
<https://github.com/notifications/beacon/ABYNHYZ7ZKLYD5HRJEHXDLLYEOANXA5CNFS
M6AAAAAA2ELLJG6WGG33NNVSW45C7OR4XAZNRIRUXGY3VONZWS33OINXW23LFNZ2KUY3PNVWWK3T
UL5UWJTQAON2HS.gif> Message ID:
***@***.***>
